### ISSUE TITLE
Socket Options don't work in newest devel branch

### ISSUE THREAD

**Issue by VirtualSmile** [z3APA3A/3proxy#277]
Hello!

I'm unable to add -os/oc/ol/or/oR command line options in the newest devel build. Both from github and from the 3proxy website. None of the socket options work and give me an error code 40. 

It works on the 18th of April Devel Build, but not on the newest one (0.9b). The only difference I can see is in proxymain.c. Between lines 135 and 139 the "setsockopt" changes from

```
void setopts(SOCKET s, int opts){
	int i, opt, set;
	for(i = 0; opts >= (opt = (1<<i)); i++){
		set = 1;
		if(opts & opt) setsockopt(s, *sockopts[i].optname == 'T'? IPPROTO_TCP:SOL_SOCKET, sockopts[i].opt, (char *)&set, sizeof(set));
	}
}
```

to 

```
void setopts(SOCKET s, int opts){
	int i, opt, set;
	for(i = 0; opts >= (opt = (1<<i)); i++){
		set = 1;
		if(opts & opt) setsockopt(s, *sockopts[i].optname == 'T'? IPPROTO_TCP:
#ifdef SOL_IP
			*sockopts[i].optname == 'I'? SOL_IP: 
#endif
			SOL_SOCKET, sockopts[i].opt, (char *)&set, sizeof(set));
	}
}
```
Which I think might be causing the problem. Could you please take a look at it? I'll try to compile it with the original version later. Thank you!

**Comment by z3APA3A** [comment]
What do you mean by "error code 40"?

**Comment by z3APA3A** [comment]
And what exactly are you trying to do?

**Comment by VirtualSmile** [comment]
Sorry let me elaborate! I'm trying to launch a socks5 proxy with this commandline:
```socks -p1709 -u2 -4 -ocTCP_NODELAY -osTCP_NODELAY -olTCP_NODELAY -orTCP_NODELAY -oRTCP_NODELAY```

However, I'm getting this error when I try to start 3proxy. This is on both Linux (Debian) and Windows 10.
```
Service not started on line: 16
Command: 'socks' failed with code 40, line 16
```
If I compile and use the older version of 3proxy (18th of April), it works with the exact same config file.

**Comment by z3APA3A** [comment]
Fixed with 68823c29211b824cb1f7e24b3ae189132e542761 in devel
